From 8637bfb1b79d7778447d93349f14b86dea41d2a4 Mon Sep 17 00:00:00 2001
From: Jan Beich <jbeich@FreeBSD.org>
Date: Sat, 23 Sep 2023 12:31:37 +0000
Subject: [PATCH] build: Unbreak build without precompiled headers (#3400)

* helpers: add missing header after 2e34548aea5b

src/helpers/VarList.cpp: In constructor 'CVarList::CVarList(const std::string&, size_t, char, bool)':
src/helpers/VarList.cpp:19:34: error: 'removeBeginEndSpacesTabs' was not declared in this scope
   19 |             m_vArgs.emplace_back(removeBeginEndSpacesTabs(in.substr(pos)));
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
src/helpers/VarList.cpp:23:30: error: 'removeBeginEndSpacesTabs' was not declared in this scope
   23 |         m_vArgs.emplace_back(removeBeginEndSpacesTabs(std::string_view{s}.data()));
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~

* helpers: add missing C linkage after 0dbd99700319

ld.lld: error: undefined symbol: wlr_region_scale(pixman_region32*, pixman_region32 const*, float)
>>> referenced by Region.cpp
>>>               src/Hyprland.p/helpers_Region.cpp.o:(CRegion::scale(float))
>>> did you mean: extern "C" wlr_region_scale
>>> defined in: /usr/lib/libwlroots.so.12032
---
 src/helpers/Region.cpp  | 2 ++
 src/helpers/VarList.cpp | 1 +
 2 files changed, 3 insertions(+)

diff --git a/src/helpers/Region.cpp b/src/helpers/Region.cpp
index e2db466438..5515e71e7e 100644
--- a/src/helpers/Region.cpp
+++ b/src/helpers/Region.cpp
@@ -1,6 +1,8 @@
 #include "Region.hpp"
+extern "C" {
 #include <wlr/util/box.h>
 #include <wlr/util/region.h>
+}
 
 CRegion::CRegion() {
     pixman_region32_init(&m_rRegion);
diff --git a/src/helpers/VarList.cpp b/src/helpers/VarList.cpp
index c29a1e9641..780ea9a045 100644
--- a/src/helpers/VarList.cpp
+++ b/src/helpers/VarList.cpp
@@ -1,3 +1,4 @@
+#include "MiscFunctions.hpp"
 #include "VarList.hpp"
 #include <ranges>
 #include <algorithm>
